beacon-chain/core/blocks/withdrawals.go

```diff
@@ -206,3 +210,32 @@ func BLSChangesSignatureBatch(
 	}
 	return batch, nil
 }
+
+// VerifyBLSChangeSignature checks the signature in the SignedBLSToExecutionChange message.
+// It validates the signature with the Capella fork version if the passed state
+// is from a previous fork.
+func VerifyBLSChangeSignature(
+	st state.BeaconState,
+	change *ethpbv2.SignedBLSToExecutionChange,
+) error {
+	var epoch types.Epoch
+	var fork *ethpb.Fork
+	if st.Version() < version.Capella {
```

Comment:
Duplicate logic. Would refactor into a small func that gives us the signing domain for a bls to exec change based on a beacon state

Reply:
Terence merged something like this yesterday I think. I'll try to adapt his helper